components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -184,6 +184,12 @@ void AtmosphereOutput::run (const std::string& filename, const bool is_write_ste
 
   using namespace scream::scorpio;
 
+  // Update all diagnostics, we need to do this before applying the remapper
+  // to make sure that the remapped fields are the most up to date.
+  for (auto const& name : m_fields_names) {
+    update_field(name); // Note, if the field is a diagnostic this will update it.
```

Comment:
Can we make this more self explanatory? Namely:
- move the if statement in here and out of the `update_field` function
- rename to `compute_diagnostic` or something that makes it clear we are only processing a diag field

So something like this
```
for (...) {
  auto it = m_diagnostic.find(name);
  if (it != m_diagnostics.end()) {
    compute_diagnostic(*it);
  }
}
```

Reply:
would it make more sense to just cycle through `m_diagnostics` rather than all of `m_field_names` and then check if they are a diagnostic?

Reply:
Yes, that's fine too.